src/runtime_src/driver/zynq/user/shim.cpp

```diff
@@ -143,144 +126,160 @@ ZYNQShim::~ZYNQShim()
 }
 #endif
 
+void ZYNQShim::mapKernelControl(std::vector<uint64_t> cus_offset) {
+  void *ptr = NULL;
+  if (cus_offset.size() == 0) {
+    return;
+  }
+
+  auto cu = cus_offset.begin();
 
-//TODO: UKP: return definition is not defined.
-size_t ZYNQShim::xclWrite(xclAddressSpace space, uint64_t offset, const void *hostBuf, size_t size)
-{
-  if(!hostBuf) {
+  while (cu != cus_offset.end()) {
+    auto it = mKernelControl.find(*cu);
+    if (it == mKernelControl.end()) {
+      ptr =
+          mmap(0, 0x10000, PROT_READ | PROT_WRITE, MAP_SHARED, mKernelFD, *cu);
+      mKernelControl[*cu] = (uint32_t *)ptr;
```

Comment:
You can reuse the iterator from the lookup which makes the insertion more efficient:
mKernelControl.insert(it, std::pair<char,int>(*cu, (uint32_t *)ptr));

Reply:
Got it. This has been done after resolved conflict.